wit/consumer.wit

```diff
@@ -1,10 +1,26 @@
-/// An interface for a generic consumer.
 default interface consumer {
-	use pkg.types.{broker, channel, subscription-token, error}
+    use pkg.types.{client, message, channel, error, guest-configuration}
 
-	/// Subscribes to a channel in a broker.
-	subscribe: func(b: broker, c: channel) -> result<subscription-token, error>
+    /// Blocking receive for t-milliseconds with ephemeral subscription – if no message is received, returns None
+    subscribe-try-receive: func(c: client, ch: channel, t-milliseconds: u32) -> result<option<list<message>>, error>
 
-	/// Unsubscribes from a channel via subscription token.
-	unsubscribe: func(b: broker, st: subscription-token) -> result<_, error>
-}
\ No newline at end of file
+    /// Blocking receive until message with ephemeral subscription
+    subscribe-receive: func(c: client, ch: channel) -> result<list<message>, error>
+
+    /// 'Fit-all' type function for updating a guest's configuration – this could be useful for:
+    ///     - unsubscribing from a channel,
+    ///     - checkpointing,
+    ///     - etc..
+    update-guest-configuration: func(gc: guest-configuration) -> result<_, error>
+
+    /// A message can exist under several statuses:
```

Comment:
This feels like an inconsistency in the abstraction level. When I think about the goals of these interfaces, one of them is typically to hide certain implementation details from people developing the components. 

What worries me about having `complete` and `abandon` is that it is surfacing an ack/nack pattern that can be problematic. Firstly, not all brokers implementing this interface may support acks and nacks at all. Secondly, some brokers may support ack/nack only under certain conditions and in certain configurations, creating unpredictability in how an application may perform from one provider to another or even from one environment to another.

Item 2 in this list even uses the word `queue`, which may or may not be how the implementer of this interface manages messages.

Put more simply: as a developer using wasi-messaging, I don't want to care about acks or nacks. I want to publish messages, make requests, and receive messages from subscriptions.